includes/functions.php

```diff
@@ -95,3 +95,25 @@ function get_self_url($strip_query=true){
     }
     return $self_url.$_SERVER['HTTP_HOST'].$url;
 }
+
+//Adapted from http://www.10stripe.com/articles/automatically-generate-table-of-contents-php.php
+function generate_toc($html_string, $depth){
```

Comment:
If it is okay I will stick with the current one, because it is simpler and I don't want to have the parameters as part of the ToC. But I can replace it if you think it is more useful.